tests/nnstreamer_plugins/unittest_plugins.cpp

```diff
@@ -2992,6 +2993,352 @@ TEST (test_tensor_filter, reopen_tflite_02_p)
+
+  ASSERT_TRUE (g_file_test (test_model, G_FILE_TEST_EXISTS));
+
+  const gchar *model_files[] = {
+    test_model, NULL,
+  };
+
+  /* prepare properties */
+  prop = g_new0 (GstTensorFilterProperties, 1);
+  ASSERT_TRUE (prop != NULL);
+
+  prop->fwname = fw_name;
+  prop->model_files = model_files;
+  prop->num_models = 1;
+
+  ASSERT_TRUE (fw && fw->open && fw->close && fw->reloadModel);
+
+  /* open tf-lite model */
+  EXPECT_TRUE (fw->open (prop, &private_data) == 0);
+
+  g_free (test_model);
+  test_model = g_build_filename (root_path, "tests", "test_models", "models",
+      "mobilenet_v2_1.0_224_quant.tflite", NULL);
+  ((gchar **)model_files)[0] = test_model; /* remove const for the test */
+
+  /* reload tf-lite model */
+  EXPECT_TRUE (fw->reloadModel (prop, &private_data) == 0);
+
+  g_free (test_model);
+  test_model = g_build_filename (root_path, "tests", "test_models", "models",
+      "model_does_not_exist.tflite", NULL);
+  ((gchar **)model_files)[0] = test_model; /* remove const for the test */
+
+  /* reload tf-lite model which does not exist */
+  EXPECT_FALSE (fw->reloadModel (prop, &private_data) == 0);
+
+  /* close tf-lite model */
+  fw->close (prop, &private_data);
+
+  g_free (prop);
+  g_free (test_model);
+}
+
+/**
+ * @brief Test to reload tf-lite; model has wrong dimension (negative)
+ */
+TEST (test_tensor_filter, reload_tflite_model_wrong_dims_n)
+{
+  const gchar fw_name[] = "tensorflow-lite";
+  const GstTensorFilterFramework *fw = nnstreamer_filter_find (fw_name);
+  GstTensorFilterProperties *prop = NULL;
+  gpointer private_data = NULL;
+
+  const gchar *root_path = g_getenv ("NNSTREAMER_BUILD_ROOT_PATH");
+  gchar *test_model;
+
+  /* supposed to run test in build directory */
+  if (root_path == NULL)
+    root_path = "..";
+
+  test_model = g_build_filename (root_path, "tests", "test_models", "models",
+      "mobilenet_v1_1.0_224_quant.tflite", NULL);
+
+  ASSERT_TRUE (g_file_test (test_model, G_FILE_TEST_EXISTS));
+
+  const gchar *model_files[] = {
+    test_model, NULL,
+  };
+
+  /* prepare properties */
+  prop = g_new0 (GstTensorFilterProperties, 1);
+  ASSERT_TRUE (prop != NULL);
+
+  prop->fwname = fw_name;
+  prop->model_files = model_files;
+  prop->num_models = 1;
+
+  ASSERT_TRUE (fw && fw->open && fw->close && fw->reloadModel);
+
+  /* open tf-lite model */
+  EXPECT_TRUE (fw->open (prop, &private_data) == 0);
+
+  g_free (test_model);
+  test_model = g_build_filename (root_path, "tests", "test_models", "models",
+      "add.tflite", NULL); /* it has unmatched dimension with mobilenet v1 */
+  ((gchar **)model_files)[0] = test_model; /* remove const for the test */
+
+  /* reload tf-lite model with unmatched dims */
+  EXPECT_FALSE (fw->reloadModel (prop, &private_data) == 0);
+
+  /* close tf-lite model */
+  fw->close (prop, &private_data);
+
+  g_free (prop);
+  g_free (test_model);
+}
+
+/**
+ * @brief Test to reload tf-lite; model has wrong dimension (negative)
```

Comment:
I think this test is for reload model file does not exist.

Reply:
Oh, wrong comment. will be updated